robottelo/libvirt_discovery.py

```diff
@@ -0,0 +1,197 @@
+    on the server consuming hardware resources.
+
+    It is possible to customize the ``libvirt_host`` and ``image_dir``
+    as per virtual machine basis. Just set the expected values when
+    instantiating.
+
+    """
+
+    def __init__(
+            self, cpu=1, ram=1024, boot_iso=False, libvirt_server=None,
+            image_dir=None, mac=None, bridge=None):
+        self.cpu = cpu
+        self.ram = ram
+        if libvirt_server is None:
+            self.libvirt_server = settings.compute_resources.libvirt_hostname
+        else:
+            self.libvirt_server = libvirt_server
+        if self.libvirt_server is None or self.libvirt_server == '':
+            raise LibvirtGuestError(
+                'A libvirt server must be provided. Make sure to fill '
+                '"libvirt_server" on compute_resources section of your '
+                'robottelo configuration. Or provide a not None '
+                'libvirt_server argument.'
+            )
+        if image_dir is None:
+            self.image_dir = settings.compute_resources.libvirt_image_dir
+        else:
+            self.image_dir = image_dir
+        if mac is None:
+            self.mac = gen_mac(multicast=False, locally=True)
+        else:
+            self.mac = mac
+        if bridge is None:
+            self.bridge = settings.vlan_networking.bridge
+        else:
+            self.bridge = bridge
+        if self.bridge is None or self.bridge == '':
+            raise LibvirtGuestError(
+                'A bridge name must be provided. Make sure to fill '
+                '"bridge" on vlan_networking section of your robottelo '
+                'configuration. Or provide a not None bridge '
+                'argument.'
+            )
+        self.boot_iso = boot_iso
+        self.hostname = None
+        self.ip_addr = None
+        self._domain = None
+        self._created = False
+        self.guest_name = 'mac{0}'.format(self.mac.replace(':', ""))
+
+    def create(self):
+        """Creates a virtual machine on the libvirt server using
+        virt-install
+
+        :raises robottelo.vm.LibvirtGuestError: Whenever a virtual guest
+            could not be executed.
+
+        """
+        if self._created:
+            return
+
+        command_args = [
+            'virt-install',
+            '--hvm',
+            '--network=bridge:{vm_bridge}',
+            '--mac={vm_mac}',
+            '--name={vm_name}',
+            '--ram={vm_ram}',
+            '--vcpus={vm_cpu}',
+            '--os-type=linux',
+            '--os-variant=rhel7',
+            '--disk path={image_name},size=8',
+            '--noautoconsole',
+        ]
+
+        if not self.boot_iso:
+            # Required for PXE-based host discovery
+            command_args.append('--pxe')
+        else:
+            # Required for PXE-less host discovery, where we boot the host
+            # with bootable discovery ISO
+            self.boot_iso_name = settings.discovery.discovery_iso
```

Comment:
Please note that `robottelo-ci` and `robottelo.properties.sample` still needs to be updated with `discovery_iso` parameter.  Idea here is to share the approach. 
I'll update robottelo-ci and sample-properties file, If all changes defined in this PR looks good to reviewers.